cmd/checkMigrations_test.go

```diff
@@ -0,0 +1,130 @@
+			oldMigrationList := []string{}
+
+			Expect(cmd.NewMigrations(newMigrationList, oldMigrationList)).To(Equal([]string{}))
+		})
+
+		It("returns no new migrations when the migration lists are identical", func() {
+			newMigrationList := []string{"one.sql"}
+			oldMigrationList := []string{"one.sql"}
+
+			Expect(cmd.NewMigrations(newMigrationList, oldMigrationList)).To(Equal([]string{}))
+		})
+
+		It("returns any migrations that are in the new, but not the old", func() {
+			newMigrationList := []string{"one.sql"}
+			oldMigrationList := []string{}
+
+			Expect(cmd.NewMigrations(newMigrationList, oldMigrationList)).To(Equal([]string{"one.sql"}))
+		})
+
+		It("ignores oldMigrations", func() {
+			newMigrationList := []string{"one.sql"}
+			oldMigrationList := []string{"two.sql"}
+
+			Expect(cmd.NewMigrations(newMigrationList, oldMigrationList)).To(Equal([]string{"one.sql"}))
+		})
+
+		It("keeps the list to unique entries", func() {
+			newMigrationList := []string{"one.sql", "one.sql"}
+			oldMigrationList := []string{"two.sql", "two.sql"}
+
+			Expect(cmd.NewMigrations(newMigrationList, oldMigrationList)).To(Equal([]string{"one.sql"}))
+		})
+	})
+
+	Describe("CheckNewMigrations", func() {
+		It("is not an error to have no new migrations", func() {
+			originalMigrations := []string{}
+			newMigrations := []string{}
+
+			err := cmd.CheckNewMigrations(originalMigrations, newMigrations)
+
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("is an error if the new migrations are out of order", func() {
+			originalMigrations := []string{"20200429072513_last.sql"}
+			newMigrations := []string{"20200429072512_newest.sql"}
+
+			err := cmd.CheckNewMigrations(originalMigrations, newMigrations)
+
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("does not change the new migrations passed in", func() {
+			originalMigrations := []string{"20200429072513_last.sql"}
+			newMigrations := []string{"20200429072515_newest.sql", "20200429072512_newest.sql"}
+
+			err := cmd.CheckNewMigrations(originalMigrations, newMigrations)
+
+			Expect(err).To(HaveOccurred())
+			Expect(newMigrations).To(Equal([]string{"20200429072515_newest.sql", "20200429072512_newest.sql"}))
+		})
+
+		It("does not erroneously report an error if the new Migrations come in out of order", func() {
+			originalMigrations := []string{"20200429072513_last.sql"}
+			newMigrations := []string{"20200429072517_newest.sql", "20200429072515_newest.sql"}
+
+			err := cmd.CheckNewMigrations(originalMigrations, newMigrations)
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(newMigrations).To(Equal([]string{"20200429072517_newest.sql", "20200429072515_newest.sql"}))
+		})
+
+		It("requires every new migration to start with a 14 digit timestamp", func() {
+			originalMigrations := []string{"20200429072513_last.sql"}
+			newMigrations := []string{"20200429072517_newest.sql", "2020042907251z_newest.sql"}
+
+			err := cmd.CheckNewMigrations(originalMigrations, newMigrations)
```

Comment:
It shouldn't - because that will number everything and should still be fine. 

`goose fix` could be problematic BTW - because it doesn't change the database for what migrations were already run. I learned that while starting this story.